app/src/main/java/com/conupods/OutdoorMaps/View/MapsActivity.java

```diff
@@ -103,8 +113,14 @@ public void onMapReady(GoogleMap googleMap) {
             createLocationRequest();
         }
 
-        MapInitializer mapInitializer = new MapInitializer(mCameraController, mMap, mBuildingInfoWindow);
-        mapInitializer.initializeSearchBar((EditText) findViewById(R.id.search));
+
+        MapInitializer mapInitializer = new MapInitializer(mCameraController, mMap,  mBuildingInfoWindow);
+
+        mSearchBar = (SearchView) mapInitializer.initializeSearchBar((SearchView) findViewById(R.id.searchBar), this);
+        setSearchViewOnClickListener(mSearchBar, (View v) -> {
```

Comment:
move this event out map activity if possible - same as what I did with setOnMarkerClickListener which is inside MapInitializer

Reply:
Moved